### PR TITLE
Feat: Adds Haskell 009 010 012

### PR DESCRIPTION
**Pull request by caiopo**
### Performance

[009 - Try it online!](https://tio.run/##fZBLS8QwFIX3@RVn4aLVtnQG3Ax2FiKCIrhQcFE6cPtwGkzTktwuBvzvNZ0OPiBjSELu5RzOl9uS/WiUmibZDb1h3BFT8iQtI3iXug6FGA7c0r43DelXIwfVMDYbPGhu9oYUNLItAh1h3iHiLW77XvlcAUUoI1QhMgG3aLfGFUp3Zxmq3VqIblTstId7aSyfDcu/04o5TvttFela1sSNPcXNzcfR/uHA/Ed4WH/cR6/UIFyidKcSgheRfZPcvozdKk3Tfym9hgUqP8EcW7ELiVGG@HSPmxj5Kkmu07Rwit9l4QZFjmiOfEYQLlWGwUjNuIB3HB4EMU1f "Haskell – Try It Online")

[010 - Try it online!](https://tio.run/##ZU@7DoJAEOzvK6awAKPGYGfAws5KCxMLJfHUVYgHd7mDiIn/jseBr7jV7M7OzG7CzZWEqGul04wMplNsF3lBF9Ix62YRAthxhQcqhENsJ6NRPEBqHI8qZm9od7kQ8HYKw5lt95k87aEww9iHd@bHQmqzlmt9R@Uz2LolpMmhpn43rFvBr7RJUkEvT9VXCCOrRnsdY6bMlueVa@Yk5K35oXuhEXTwf@0gy/yEyGVbEr3vsLClPykZT3PnvITnt13UkHlhhX/ewdgVq@sn "Haskell – Try It Online")

[012 - Try it online!](https://tio.run/##jVA9T8MwFNz9K27oYIs2SpFYKlIJqQuolAWJIWRwWyexcJxiv0CR@O/BST/UAhJ4sGzdvXt3V0r/ooxpW11takeYSZLRXHsCz7VdD2GbpWCn4L38WKqAurq6azwJxshpaQujFk21VM5jMkF6a0kVymU/wAR@Ja0BvxAYI72MooyxtX7Tvt6N7icxmp6oHAk2CPAxJgKDzlq4V7VdSUKabocItMDYZvjEFtejXt7oSlM2hFNVhyFJEGcM3XkvlVPoCUE2N3XtEMG/hqgDdAH7/U4aWMZy7Tw97tM8aSpvaK6kp8Xsd/P75z/mQqbez6HSbnnoHvz5UF6nx88MDWCULagMj2M3B7bANIEV@FY9Y5XUtnf5AC52vwQbp@1u5V8@r@KYte0X "Haskell – Try It Online")

### CODE REVIEW

File: src/009/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-https://projecteuler.net/problem=004
+https://projecteuler.net/problem=009
```
Comment (caian-gums) [inline src/009/README.md]: Nice catch!

**Comment by caian-gums** [comment]
Hey, @caiopo you are on fire! Is nice that you are feeding us with some Haskell code!

As my point of view, we are trying to maintain some guidelines on our code here and as most, as I think that create verbose functions names is a good thing (Haskell use functions, right?) @FrankKair changed some of our code to keep this pattern. The pattern is: You declare your 'solving' function with the `solve` name, as all the code on the `009`, `010`, ... directories solve the specific Euler Problem described on the `README.md` file in that directory.

@FrankKair can say if we'll keep using this pattern but for me it's all good and I'll approve your PR but wait for the @FrankKair merge!

**Comment by caiopo** [comment]
I don't really like calling functions generic names like `solve` or `calculate` as they carry little context about the functions purpose.

I know some names (like `firstTriangleWithAtLeastNDivisors`) can seem a bit too much verbose, but they convey exactly what that function does.

File: src/009/README.md (hunk 1 above)
Comment (FrankKair) [reply src/009/README.md]: Nice one, indeed 👍🏼

**Comment by caian-gums** [comment]
Just to be clear @caiopo: I like verbose function names and I personally don't mind that they are long(actually, I prefer). `Swift` has even variables with long names and I think this helps the reading of the code.

As I said, even not knowing how to program `Haskell` I can read and understand what each function you wrote. And this is great! My only concern is about the naming and convention that I saw @FrankKair trying to maintain.

Congrats! 🎉

**Pull request merged**